opencog/guile/SchemeSmob.cc

```diff
@@ -45,6 +47,25 @@ void SchemeSmob::init()
 	if (is_inited.test_and_set()) return;
 
 	init_smob_type();
+
+	// If one is importing this class then the functions defined in
+	// .scm files associated with '(opencog)' module should be available.
+	// Assuming installation in a standard path.
+	scm_c_eval_string("(add-to-load-path \"/usr/local/share/opencog/scm\")\n");
+
+	// Add paths set through GUILE_LOAD_PATH
+	std::string env_str;
+
+	if (const char* _env = std::getenv("GUILE_LOAD_PATH")) {
+		env_str = _env;
+		auto output = StringManipulator::split(env_str, ":");
+
+		for (const auto& i : output) {
+			std::string scm_command = "(add-to-load-path \"" + i + "\")";
+			scm_c_eval_string(scm_command.c_str());
+		}
+	}
```

Comment:
I like this, but it would be better to put it here: https://github.com/opencog/atomspace/blob/master/opencog/guile/load-file.cc#L82  which is the set of paths currently being searched.

Or better yet, maybe `load_scm_file_relative()` should be completely removed, and replaced by `scm_c_primitive_load_path()` and by this add-to-load-path mechanism.